app/forms/mappings/CrnRegex.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.mappings
+
+import play.api.libs.json.{Reads, Writes}
+import uk.gov.hmrc.domain._
+
+case class CrnRegex(crn: String) extends TaxIdentifier with SimpleName {
+
+  val name = "crn"
+  require(CrnRegex.isValid(crn), s"$crn is not a valid company registration number.")
+
+  override def toString = crn
+  
+  def value = crn
+}
+
+object CrnRegex extends (String => CrnRegex) {
```

Comment:
As above this seems to only exist for the regex to be used in the form. Should we just put it in the form instead? There's a lot of confusing stuff going on here.